gnovm/pkg/gnolang/type_check.go

```diff
@@ -1029,6 +1029,14 @@ func assertValidAssignRhs(store Store, last BlockNode, n Node) {
 			tt = evalStaticType(store, last, exp)
 			panic(fmt.Sprintf("%s (type) is not an expression", tt.String()))
 		}
+
+		// Ensures that function used in ValueDecl or AssignStmt must return at least 1 value.
+		if cx, ok := exp.(*CallExpr); ok {
+			tType, ok := tt.(*tupleType)
+			if ok && len(tType.Elts) == 0 {
```

Comment:
do you think the same logic in `evalStaticTypeOf` is still needed?

Reply:
IMHO, I think the the login in evalStaticTypeOf and the check in this PR do not target the same check so they should exist both

Reply:
Ah, maybe I misunderstood your comment, did you mean that should we have the same check in `evalStaticTypeOf ` ?

Reply:
what's the logic in `evalStaticTypeOf` targeting? I mean if it's not needed we can just remove it, please check, thank you.

Reply:
I've checked, it should be fine to remove the check in `evalStaticTypeOf`. Thanks